src/runtime/job.h

```diff
@@ -19,15 +19,34 @@
 #define JOB_H
 
 #include <memory>
+#include <string>
 
 struct Database;
 struct Runtime;
 
+struct ResourceBudget {
+  double percentage;
+  uint64_t fixed;
+
+  static const char *parse(const char *str, ResourceBudget &output);
```

Comment:
It's a bit unintuitive, at least to me, that this returns a string error message on failure or a nullptr otherwise just based on its type signature, so could that be documented in a short comment here?

Reply:
Done